core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -57,6 +58,18 @@ public void onCreated(Item item) {
     }
 
     /**
+     * Called before a job is copied into a new parent, providing the ability to veto the copy operation before it
+     * starts.
+     *
+     * @param src the item being copied
+     * @param parent the proposed parent
+     * @throws Failure to veto the operation.
+     * @since TODO
+     */
+    public void onCheckCopy(Item src, ItemGroup parent) throws Failure {
```

Comment:
"Represents an error induced by user, encountered during HTTP request processing."

translation, this represents an error - resulting from a user action - that occurred while the HTTP request of the user was being processed.

Here we have a user request - to copy something - and we cannot complete the HTTP request.

Further, the other checks in the call path all throw a `Failure`, e.g. https://github.com/jenkinsci/jenkins/blob/01d793154617a2bbccc00ceea11b457334d16a90/core/src/main/java/hudson/model/ItemGroupMixIn.java#L169

So we are left with your only objecting being that it is a `RuntimeException`

Seems overly reaching to require code duplication just to make the exception non-runtime exception